File: client/src/sim/combatMath.ts
export const computeDamage = (attack: number, defense: number) => {
  const mitigated = attack - defense * 0.5
  return Math.max(1, Math.floor(mitigated))
}

export const applyVariance = (baseDamage: number, variance: number, roll: number) => {
  const spread = baseDamage * variance
  return Math.max(1, Math.floor(baseDamage + (roll - 0.5) * spread))
}

File: Vampires & Werewolves/Assets/Scripts/Combat/CombatMath.cs
using UnityEngine;

public static class CombatMath
{
    public static int ComputeDamage(float attack, float defense)
    {
        float mitigated = attack - defense * 0.5f;
        return Mathf.Max(1, Mathf.FloorToInt(mitigated));
    }

    public static int ApplyVariance(int baseDamage, float variance, float roll)
    {
        float spread = baseDamage * variance;
        return Mathf.Max(1, Mathf.FloorToInt(baseDamage + (roll - 0.5f) * spread));
    }
}

